pkg/storage/unified/resource/search_queue.go

```diff
@@ -128,7 +134,10 @@ func (b *indexQueueProcessor) process(batch []*WrittenEvent) {
 		req.Items = append(req.Items, item)
 	}
 
+	b.mu.Lock()
 	err := b.index.BulkIndex(req)
+	b.mu.Unlock()
```

Comment:
This keeps the lock for very long time (for entire indexing), which can block goroutine running `dispatchEvent` method from processing further events until indexing of previous event for the same NamespacedResource is finished.

Can we instead take the index with the lock, and then run indexing without the lock?

```go
    b.mu.Lock()
    idx := b.index
    b.mu.Unlock()

    idx.BulkIndex(req)
```